pkg/builder/container_builder.go

```diff
@@ -17,13 +17,42 @@ limitations under the License.
 package builder
 
 import (
+	"crypto/sha256"
+	"encoding/hex"
 	"encoding/json"
 	"strings"
 
 	corev1api "k8s.io/api/core/v1"
 	apimachineryRuntime "k8s.io/apimachinery/pkg/runtime"
 )
 
+// maxContainerNameLen is the maximum allowed length for Kubernetes container names
+// to satisfy DNS-1123 label constraints (RFC-1123)
+const maxContainerNameLen = 63
+
+// hashString generates a SHA256 hash of the input string for creating unique identifiers
+func hashString(s string) string {
+	hash := sha256.Sum256([]byte(s))
+	return hex.EncodeToString(hash[:])
+}
+
+// ensureValidNameLength ensures the container name fits DNS-1123 label constraints (max 63 chars).
+// If the name exceeds the limit, it truncates and appends an 8-character hash suffix to maintain uniqueness.
+func ensureValidNameLength(base string) string {
```

Comment:
No need to introduce a new function to truncate the name to 63 characters.
There are already two functions that do the same job.
https://github.com/vmware-tanzu/velero/blob/a1026cb5319a9b543d5f9ee2f2e349578b9c4bb8/pkg/label/label.go#L32-L62